mesh/mesh.go

```diff
@@ -239,23 +232,66 @@ func (msh *Mesh) GetLayer(i types.LayerID) (*types.Layer, error) {
 	return l, nil
 }
 
+type internalLayer struct {
+	layerID   types.LayerID
+	layerHash types.Hash32
+}
+
 type validator struct {
 	*Mesh
-	processedLayer types.LayerID
+	processedLayer internalLayer
 }
 
 func (vl *validator) ProcessedLayer() types.LayerID {
-	defer vl.lvMutex.RUnlock()
-	vl.lvMutex.RLock()
+	return vl.getProcessedLayer().layerID
+}
+
+func (vl *validator) ProcessedLayerHash() types.Hash32 {
+	return vl.getProcessedLayer().layerHash
+}
+
+func (vl *validator) SetProcessedLayer(layerID types.LayerID, hash types.Hash32) {
```

Comment:
this function is used only internally and can be unexported

Reply:
done